cardverificationsheet/src/main/java/com/stripe/android/cardverificationsheet/framework/util/ItemCounter.kt

```diff
@@ -0,0 +1,37 @@
+package com.stripe.android.cardverificationsheet.framework.util
+
+import androidx.annotation.CheckResult
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+/**
+ * A class that counts and saves items.
+ */
+class ItemCounter<T>(initialValue: T) {
+    private val storageMutex = Mutex()
```

Comment:
maybe easier to implemented with a `mutableMapOf<T, AtomicInteger>` ? that way you don't need mutex

Reply:
I looked into this originally. The problem here is setting the initial value for a given key. `ConcurrentHashMap` would work, except for two issues:

1. `ConcurrentHashMap.compute` is not supported at API < 24 (we require 21)
2. `ConcurrentHashMap.compute` uses `synchronized` blocks, which blocks the entire thread rather than just the coroutine that `Mutex` blocks.